torchtitan/experiments/README.md

```diff
@@ -0,0 +1,18 @@
+## Contributing Guidelines
+
+We provide this `experiments/` folder to host experiments that add significant value to `torchtitan`, with the following principles. We refer to the part of `torchtitan` outside `experiments` as `core`.
+1. Each subfolder in `experiments` will be an experiment, with a clear theme which can be flexible, such as
+    - a new model, or preferably a new model architecture, with its training infrastructure including parallelization functions;
+    - an enhancement or addition to the existing infrastructures in `torchtitan`.
+2. It is the contributors' responsibility to justify the value of an experiment. `torchtitan` team will review proposals on a case-by-case basis. As part of the contribution, the contributors should provide documentation that clearly showcases the motivation and innovation of an experiment, including reports on performance and loss convergence.
+3. An experiment should reuse existing `torchtitan` code as much as possible, such as modules in [`components/`](../components/) (via a new [`TrainSpec`](../protocols/train_spec.py)) and [`train.py`](../train.py). For a list of extension points we provide, please refer to [docs/extension.md](../../docs/extension.md).
+    - The extension points are subject to change. Contributors should provide feedback if they find troubles reusing the components, instead of simply making copy-and-paste.
+    - The degree to which existing components are reused and whether duplications are legit will also be a criteria whether an experiment would be accepted.
+4. Each experiment is independent from other experiments, and can have its own dependencies (on top of [core dependencies](../../requirements.txt)), and its own tests.
+5. The dependency from `experiments` to `core` is one-way. Anything in `experiments` is optional for `core` to run successfully. In particular, development in `core` is not blocked by breakage in `experiments`. We will utilize GitHub's [CI mechanism](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) to help test an experiment periodically and only if the experiment itself is affected by a PR.
+6. Each experiment needs to have an owner. The owner is responsible to work with `torchtitan` team to maintain the quality and healthiness of an experiment, which includes
+    - adapting an experiment to changes in `core` and fix broken tests, no late than the next official `torchtitan` release;
```

Comment:
no late**r** than the next official `torchtitan` release;   (r is missing for late)